### PR TITLE
Load IDEMIA PKCS11 driver

### PR DESCRIPTION
**Pull request by metsma**
IB-5724

Signed-off-by: Raul Metsma <raul@metsma.ee>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -35,6 +37,8 @@ elseif(UNIX)
             DESTINATION "${CMAKE_INSTALL_DATADIR}/mozilla/extensions/{ec8030f7-c20a-464f-9b0e-13a3a9e97384}/")
     install(FILES ${PROJECT_SOURCE_DIR}/onepinopenscpkcs11.json
             DESTINATION "${CMAKE_INSTALL_LIBDIR}/mozilla/pkcs11-modules")
+    install(FILES ${PROJECT_SOURCE_DIR}/idemiapkcs11.json
```
Comment (martinpaljak) [inline CMakeLists.txt]: The PKCS#11 module in question is for/from the "awp" productline. Maybe not call it as generic as "companyname" as there might very well be more modules from a company than one.

Comment (metsma) [reply CMakeLists.txt]: As far as I know they distribute same package to all customers.
Rename to AWP then?

File: webextension/background.js [hunk 2]
```diff
@@ -20,22 +20,20 @@
...
-      return;
+  async function load(modname) {
+    try {
+      var isInstalled = await browser.pkcs11.isModuleInstalled(modname);
+      if(isInstalled) {
+        console.log("module installed: " + modname);
+        return;
+      }
+      const PKCS11_PUB_READABLE_CERT_FLAG = 0x1<<28;
+      await browser.pkcs11.installModule(modname, PKCS11_PUB_READABLE_CERT_FLAG);
```
Comment (martinpaljak) [inline webextension/background.js:31]: I remove this extension (or I manage to uninstall this extension) if I don't like it. It leaves all the garbage behind.

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/management/onUninstalled

https://github.com/open-eid/firefox-pkcs11-loader/issues/15

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/pkcs11/uninstallModule

Comment (metsma) [reply webextension/background.js:31]: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/management/onUninstalled claims that this is not supported on firefox

**Review by martinpaljak — CHANGES_REQUESTED** [review]
If possible, I would even unload any installed modules when disabling the extension, if possible

**Pull request merged**